### ISSUE TITLE
Refactor: pay form

### ISSUE THREAD

**Issue by korhaliv** [LN-Zap/zap-desktop#2100]
## Description
Refactor pay form component. Decompose it into smaller components. Use functional component + hooks instead of class

**Comment by bolatovumar** [comment]
@korhaliv I can take a stab at this. I was wondering about the "Use functional component + hooks instead of class" bit because function components don't support proptypes AFAIK. Do we want to get rid of the existing proptypes on the Pay component and whatever components will be split out of it?

**Comment by korhaliv** [comment]
Functional components actually support proptypes without any issues. We have many instances of that in our codebase. For example
https://github.com/LN-Zap/zap-desktop/blob/next/renderer/components/UI/ActionButton.js
Also, be advised that this kind of PR should go against the `next` branch

**Comment by bolatovumar** [comment]
@korhaliv right, I see. I don't usually use proptypes and React docs show them being used with class components only so I assumed there weren't supported with function components.
